server/stream.go

```diff
@@ -4399,6 +4400,11 @@ func (mset *stream) processJetStreamMsg(subject, reply string, hdr, msg []byte,
 		}
 	}
 
+	// if we are NOT doing subject remapping, this is a message to on-board it shouldn't store Nats-* headers
+	if tsubj == _EMPTY_ {
+		hdr = removeStreamIdentityHeaders(hdr)
```

Comment:
I think this is correct, but big change since we have always stored the message and headers as we received them. Be good to discuss with @wallyqs and @bruth

Reply:
More importantly - this doesn't fix the issue for direct get - it simply solves it for the on-boarding. We would still need to have the initial fix on the outbound to ensure that clients won't get duplicate headers.

Reply:
Yes since they are stored with headers today..